mono/mini/aot-compiler.c

```diff
@@ -12728,9 +12728,13 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options,
 			else
 				acfg->llvm_sfile = g_strdup (acfg->aot_opts.llvm_outfile);
 		} else {
-			acfg->tmpbasename = (strcmp (acfg->aot_opts.temp_path, "") == 0) ?
-				g_strdup_printf ("%s", "temp") :
-				g_build_filename (acfg->aot_opts.temp_path, "temp", NULL);
+			if (strcmp (acfg->aot_opts.temp_path, "") == 0) {
+				acfg->tmpbasename = mkdtemp(g_strdup ("mono_aot_XXXXXX"));
```

Comment:
When is this deleted? Consider not using a directory if the answer is never or difficult to fix.

Reply:
Also the string needs to be writable. An array like in my earlier version, not a constant.

Reply:
Oh, you have strdup. You need to free it somewhere. Maybe you do already.

Reply:
Strdup is overkill. Just use a local array.

Reply:
This looks like a leak. Consider this instead:
char temp [] = "mono_aot_XXXXXX";
mkdtemp(temp)..

however other paths also leak, so g_free for all of them is the likely fix.